torch/fx/experimental/symbolic_shapes.py

```diff
@@ -2794,6 +2885,13 @@ def issue_guard(guard: ShapeGuard) -> None:
         if _translation_validation_enabled():
             from torch.fx.experimental.validator import PopulateValidator
 
+            # Add all deferred runtime assertions; these are not technically
+            # handled by produce_guards but we need to put them in the target
+            # set
+            for ras in self.deferred_runtime_asserts.values():
+                for ra in ras:
+                    self._add_target_expr(ra.expr)
```

Comment:
@ysiraichi Here's the fix. I can't easily test it in this PR as there are no uses of expect_true, but I'm able to show in the next PR that TV fails without this change and passes after on this test: https://github.com/pytorch/pytorch/pull/106788/files#diff-0daa64329e2d8648fc119f7809dc810c744508d345ea023116614ccc17e57dbe